barcode_utils.py

```diff
@@ -0,0 +1,166 @@
+from barcode import *
+
+import base64
+import struct
+import skip32
+import code128
+import os
+
+
+# TODO: currently broken, need to rewrite in the context of uber
+def generate_barcode_csv(filename):
+    badge_types = self.badge_types['badge_types']
+
+    # TODO: modify so we read the config directly from uber.
+    lines = []
+    for badge_type, ranges in badge_types.items():
+        range_start = int(ranges['range_start'])
+        range_end = int(ranges['range_end'])
+        lines = lines + self.generate_barcode_nums(range_start, range_end)
+
+    f = open(filename,'w')
+
+    for line in lines:
+        f.write(line + os.linesep)
+
+    f.close()
+
+
+# TODO: currently broken, need to rewrite in the context of uber
+def generate_barcode_nums(range_start, range_end):
+    generated_lines = []
+    seen_barcodes = []
+    for badge_num in range(range_start, range_end+1):
+        barcode_num = generate_barcode_from_badge_num(
+            badge_num=int(badge_num),
+            event_id=self.event_id,
+            salt=self.salt,
+            key=self.secret_key
+        )
+
+        line = "{badge_num},{barcode_num}".format(
+            badge_num=badge_num,
+            barcode_num=barcode_num,
+        )
+        generated_lines.append(line)
+
+        # ensure that we haven't seen this value before
+        if barcode_num in seen_barcodes:
+            raise ValueError('COLLISION: generated a badge# that\'s already been seen')
+        else:
+            seen_barcodes.append(barcode_num)
+
+    return generated_lines
+
```

Comment:
That's something I deliberate left out because I wanted us to write a couple of plugins first to see what kind of config values we ended up with in practice so we could make a decision based on our actual use cases instead of trying to guess in advance at what our use cases would be.

So far we have three plugins which define their own config:
- The ``magprime`` plugin defines a ``[season_pass_event]`` section.  This is something we don't care about ended up in ``c``.
- This repo defines some secret settings which we probably would want to end up on ``c`` as if they were part of the ``[secret]`` section.
- The new ``hotel`` plugin I wrote over the weekend (which I plan on making a PR for tonight) defines two options which we want to end up on ``c`` as-is and one option which we want to end up on ``c`` after some preprocessing.

So far I'm not sure the best approach.  Maybe a function call like ``c.integrate_plugin_settings(hotel_config)`` which adds those options onto ``c``?  That's starting to look like it's the best option given the use cases which we're seeing crop up.